### PR TITLE
Refactor rpc service

### PR DESCRIPTION
**Pull request by heycalmdown**
RPCService's methods must be shorter and more readable.

* introduce a new dependency - [deprecated-decorator](https://www.npmjs.com/package/deprecated-decorator)
* split files
    * `services/rpc-service` -> `utils/rpc-request`
    * `services/rpc-service` -> `utils/rpc-response`
* new features
    * when found the response consumer is canceled, RPCService will emit SIGTERM to shut down the process
    * distinguish between a timeout and an invalid response
        * store 20 timed-out requests to determine
* fixed
    * #90 - RPC error has been sent twice
    * #38 - RpcService.register should be made cleaner
* changes
    * IConsumerInfo.consumerOpts is optional
    * _publish is deprecated
* more test
    * RpcResponse
    * edge-cases
        * timeout
        * retry with 503 error
        * invoke returns a raw buffer
        * shutdown by the response queue has gone
* refactor
    * `#register` - split into multiple methods
    * `#invoke`
        * replaced setTimeout to Bluebird.timeout
        * merged reqTimeouts and reqExecutors into waitingResponse
    * `#consume` - limits its responsibility to process a raw channel only
    * merged `#invoke` and `#markTattoo` (it was very confused)
    * moved variables to near where they are used

### CODE REVIEW

File: src/services/amqp-channel-pool-service.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class AmqpChannelPoolService {
     return channel.close();
   }
 
-  usingChannel<T>(task: (channel: amqp.Channel) => PromiseLike<T>): Promise<T> {
+  async usingChannel<T>(task: (channel: amqp.Channel) => PromiseLike<T>): Promise<T> {
     return Promise.resolve(Bluebird.using(this.getChannelDisposer(), task));
```
Comment (tedlines) [inline src/services/amqp-channel-pool-service.ts]: "Promise.resolve" should be removed.
